botocore/signers.py

```diff
@@ -196,6 +208,65 @@ def generate_presigned_url(self, request_dict, expires_in=3600,
         return request.url
 
 
+class CloudFrontSigner(object):
+
+    @staticmethod
+    def b64encode(content):
+        return base64.b64encode(content).translate(maketrans(b'+=/', b'-_~'))
+
+    def __init__(self, key_id, private_key):
+        """Create a CloudFrontSigner.
+
+        :type key_id: str
+        :param key_id: The CloudFront Key Pair ID
+
+        :type private_key: str
+        :param private_key: The private key string used for signing.
+        """
+        self.key_id = key_id
+        self.rsa_pk = rsa.PrivateKey.load_pkcs1(private_key.encode('utf8'))
```

Comment:
We should ensure that the type passed into private_key is a string. i.e. reject Binary types because the encode would fail if a bytes object is passed in.